### DISCUSSION TITLE
Add more metadata info - page label and filename?

### DISCUSSION THREAD

**Discussion by vishhvak** [nlmatics/llmsherpa#22] [Q&A]
I'm trying to use it as a pdf reader for llama index, which usually also has details like page label with each document. Anyway to add that info too? How would I go about customizing it to do that myself?

**Comment by ansukla** [comment]
Yes, this is possible. We will expose the raw json at every node, so that you can get that info while indexing. Could you please share the code you are using to index?

**Comment by vishhvak** [comment]
I'm using the example from [ llama_hub ](https://github.com/run-llama/llama-hub/tree/main/llama_hub/smart_pdf_loader)

**Comment by ansukla** [comment]
Got it, will push a PR soon to add meta information to the loader.

**Comment by vishhvak** [comment]
If I weren't using the loader, how would I add metadata myself (need to get this to work asap for work :) )

**Comment by ansukla** [comment]
For now, you can simply iterate through the chunks and use the chunk_id here. page_idx and other meta info will be added to the chunks in a subsequent release.

from llama_index.readers.schema.base import Document
from llama_index import VectorStoreIndex

index = VectorStoreIndex([])
for chunk_id, chunk in enumerate(doc.chunks()):
    index.insert(Document(text=chunk.to_context_text(), extra_info={"id": chunk_id}))
query_engine = index.as_query_engine()

**Comment by vishhvak** [comment]
Is there a way to go from json to chunks? If there is, maybe I can manually add page label data that way?

**Comment by ChintanDonda** [comment]
I've the same requirement.
Can we add page number & PDF filename (or url in case of online PDF) as a metadata with each chunk?

I require this to get the SOURCE of information retrieved in the RAG-QA application.

**Comment by ansukla** [comment]
Metadata is now available via chunk.block_json or chunk.page_idx, chunk.block_idx etc.

**Comment by ansukla** [comment]
Hello, you can also access the bbox of each box now which will let you highlight relevant sections on a PDF.

**Comment by tahitimoon** [comment]
The bbox of the table has issues; the content of the current table is very long, but the bbox height is very short, as shown in the figure below.

{'chunk_type': 'table', 'page': 31, 'bbox': [438.73, 123.0, 960.6, 132.22]}

DOCUMENTS INCORPORATED BY REFERENCE > PART I > (In millions)
 |  | Years ended
 | --- | ---
 |  | September 30, 2023 | September 24, 2022 | September 25, 2021
 | --- | --- | --- | ---
 | Net income | $ 96,995 | $ 99,803 | $ 94,680
 | Other comprehensive income/(loss): Change in foreign currency translation, net of tax | (765) | (1,511) | 501
 | Change in unrealized gains/losses on derivative instruments, net of tax: Change in fair value of derivative instruments | 323 | 3,212 | 32
 | Adjustment for net (gains)/losses realized and included in net income | (1,717) | (1,074) | 1,003
 | Total change in unrealized gains/losses on derivative instruments | (1,394) | 2,138 | 1,035
 | Change in unrealized gains/losses on marketable debt securities, net of tax: Change in fair value of marketable debt securities | 1,563 | (12,104) | (694)
 | Adjustment for net (gains)/losses realized and included in net income | 253 | 205 | (273)
 | Total change in unrealized gains/losses on marketable debt securities | 1,816 | (11,899) | (967)
 | Total other comprehensive income/(loss) | (343) | (11,272) | 569
 | Total comprehensive income | $ 96,652 | $ 88,531 | $ 95,249
